code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1718,79 +1714,39 @@ pub(crate) fn do_reward_accumulation<T: Config>(
 				.saturating_mul(releasable_periods_surpassed.saturated_into::<u64>()),
 		);
 
-		// TODO(benluelo): This can probably be simplified, review the period calculations
-		if u128::saturating_add(new_total_rewards, reward.total_dilution_adjustment.into()) <=
-			reward.max_rewards.into()
-		{
-			let balance_on_hold = T::Assets::balance_on_hold(asset_id, pool_account);
-			if !balance_on_hold.is_zero() && balance_on_hold >= newly_accumulated_rewards.into() {
-				T::Assets::release(
-					asset_id,
-					pool_account,
-					newly_accumulated_rewards.into(),
-					false, // not best effort, entire amount must be released
-				)
-				.expect("funds should be available to release based on previous check; qed;");
-
-				reward.total_rewards = new_total_rewards.into();
-				reward.last_updated_timestamp = last_updated_timestamp;
-				Ok(())
-			} else {
-				Err(RewardAccumulationCalculationError::RewardsPotEmpty)
-			}
-		} else if reward.total_rewards.saturating_add(reward.total_dilution_adjustment) <
-			reward.max_rewards
-		{
-			// if the new total rewards are less than or equal to the max rewards AND the current
-			// total rewards are less than the max rewards (i.e. the newly accumulated rewards is
-			// less than the the amount that would be accumulated based on the periods surpassed),
-			// then release *up to* the max rewards
-
-			// REVIEW(benluelo): Should max_rewards be max_periods instead? Currently, the
-			// max_rewards isn't updatable, and once the max_rewards is hit, it's expected that no
-			// more rewards will be accumulated, so it's ok to not reward an entire period's worth
-			// of rewards. Review this if the max_rewards ever becomes updateable in the future.
-
-			// SAFETY(benluelo): Usage of Sub::sub: reward.total_rewards is known to be less than
-			// reward.max_rewards as per check above
-			let rewards_to_release = reward.max_rewards.sub(reward.total_rewards).into();
-
-			let balance_on_hold = T::Assets::balance_on_hold(asset_id, pool_account);
-			if !balance_on_hold.is_zero() && balance_on_hold >= newly_accumulated_rewards.into() {
-				T::Assets::release(
-					asset_id,
-					pool_account,
-					rewards_to_release.into(),
-					false, // not best effort, entire amount must be released
-				)
-				.expect("funds should be available to release based on previous check; qed;");
-
-				// return an error, but update the reward first
-				reward.total_rewards = reward.max_rewards;
-				reward.last_updated_timestamp = last_updated_timestamp;
-				Err(RewardAccumulationCalculationError::MaxRewardsAccumulated)
-			} else {
-				Err(RewardAccumulationCalculationError::RewardsPotEmpty)
+		if !total_locked_rewards.is_zero() && !newly_accumulated_rewards.is_zero() {
```

Comment:
```suggestion
		if !newly_accumulated_rewards.is_zero() {
```
Only `newly_accumulated_rewards` needs to be checked to be non-zero